spring-cloud-stream-binder-kinesis/src/test/java/org/springframework/cloud/stream/binder/kinesis/KinesisBinderTests.java

```diff
@@ -38,6 +45,21 @@
 	@ClassRule
 	public static LocalKinesisResource localKinesisResource = new LocalKinesisResource();
 
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testAutoCreateStreamForNonExistingStream() throws Exception {
+		Binder binder = getBinder();
+		RetryTemplate metatadataRetrievalRetryOperations = new RetryTemplate();
+		metatadataRetrievalRetryOperations.setRetryPolicy(new SimpleRetryPolicy());
+		FixedBackOffPolicy backOffPolicy = new FixedBackOffPolicy();
+		backOffPolicy.setBackOffPeriod(1000);
+		DirectChannel output = new DirectChannel();
+		ExtendedConsumerProperties<KinesisConsumerProperties> consumerProperties = createConsumerProperties();
+		String testTopicName = "nonexisting" + System.currentTimeMillis();
+		Binding<?> binding = binder.bindConsumer(testTopicName, "test", output, consumerProperties);
+		binding.unbind();
```

Comment:
This has not been addressed.
I'm fine with the rest of changes, but would like to see some assertion here anyway.
E.g. assert that `binding` has right properties.